templates/gitserver/gitserver.StatefulSet.yaml

```diff
@@ -68,12 +64,20 @@ spec:
           name: config-file
         name: sg-config
       - name: repos
-        persistentVolumeClaim:
-          claimName: gitserver-{{ add1 $gitserverIndex }}
       {{- if $global.Values.cluster.gitserver.ssh }}
       - name: ssh
         secret:
           defaultMode: 384
           secretName: gitserver-ssh
       {{ end }}
-{{ end }}
+  volumeClaimTemplates:
+  - metadata:
+      annotations:
+        volume.beta.kubernetes.io/storage-class: {{ $global.Values.cluster.storageClass.name }}
+      name: repos
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: {{ required ".Values.cluster.gitserver.diskSize is required" $global.Values.cluster.gitserver.diskSize }}
```

Comment:
I know we have some custom volume stuff. Just wondering if this looks good to you @beyang

Reply:
Looking at the diff, it doesn't look like we regressed on any volume stuff.

Reply:
We use manually provisioned volumes in prod, but that doesn't use this chart. AFAIK no customers use manual volumes, as the docs tells you to create a storage class.

Any volume customization would therefore be through the storage class. As long as this matches the old PVC (which this does), we're okay.